.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,57 @@
+name: Build a docker image for MintPy
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  dockerize:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set environment variables for docker build
+        run: |
+          # Lowercase repo for Github Container Registry
+          echo "REPO=${GITHUB_REPOSITORY,,}" >>${GITHUB_ENV}
+          # Ensure tags are checked out
+          git fetch origin +refs/tags/*:refs/tags/*
+          # Version number from tag
+          echo "MINTPY_VERSION=$(git describe --tags)" >> $GITHUB_ENV
```

Comment:
This creates a human readable name for a commit that looks like:
```shell
$ git describe
v1.3.1-54-g345d1dc1
# looks like: [last tag]-[distance (commits) from last tag]-g[short commit SHA]
```
if you're on a commit that's tagged, you'll just get the tag. 
```shell
$ git checkout v1.3.1
$ git describe --tags
v1.3.1
```

**Note:** This currently will look at *any* tag, even a tag like `foo`. Typically, you'd only use annotated tags for releases and `git describe` without the `--tags` option would ignore all other tags (so you can have/use non-release tags). But, MintPy has used both light-weight and annotated tags for release tags, so with this, *the only tags that should be pushed to the repo are release tags.*